python/tests/parsers/test_daily_index_crawler.py

```diff
@@ -0,0 +1,175 @@
+from datetime import date
+import httpretty
+import pytest
+import requests
+from unittest.mock import patch, MagicMock, Mock
+
+from urllib3.exceptions import ResponseError
+
+from edgar_filing_searcher.parsers.daily_index_crawler import ensure_13f_filing_detail_urls, \
+    get_subdirectories_for_specific_date, generate_dates, get_request_response
+from edgar_filing_searcher.errors import BadWebPageResponseException, InvalidUrlException
+
+from requests.exceptions import RetryError
+
+DATE_1 = date(2021, 1, 8)
+DATE_2 = date(2021, 1, 9)
+DATE_3 = date(2021, 1, 10)
+SUBDIRECTORIES = ['1478997/0001478997-21-000001',
+                  '819864/0000819864-21-000002',
+                  '1567784/0000909012-21-000002',
+                  '1479844/0001479844-21-000001']
+
+@pytest.fixture
+def filing_detail_with_no_13f_filing_urls():
+    """Creates an fixture with edgar_current_events.html data"""
+    with open("tests/fixtures/company.20210108_RemovedAllFilings.idx", "rt") as file:
+        test = file.read()
+        return test
+
+
+@httpretty.activate(allow_net_connect=False)
+def test_get_response_noError_StatusCode():
+    """Tests get_response status code when there is no error"""
+    test_url = "http://foo-api.com/data?page=2"
+    httpretty.register_uri(
+        httpretty.GET,
+        test_url,
+        body='{}',
+        status=200,
+        content_type="text/json",
+    )
+    actual = get_request_response(test_url)
+    assert actual.status_code == 200
+
+
+@httpretty.activate(allow_net_connect=False)
+def test_get_response_noError_NumberOfRequests():
+    """Tests get_response number of requests code when there is no error"""
+    test_url = "http://foo-api.com/data?page=2"
+    httpretty.register_uri(
+        httpretty.GET,
+        test_url,
+        body='{}',
+        status=200,
+        content_type="text/json",
+    )
+    get_request_response(test_url)
+    assert len(httpretty.latest_requests()) == 1
+
+
+@httpretty.activate(allow_net_connect=False)
+def test_get_response_statusError_StatusCode():
+    """Tests get_response status code when there is an error"""
+    test_url = "http://foo-api.com/data?page=2"
+    httpretty.register_uri(
+        httpretty.GET,
+        test_url,
+        body='{}',
+        status=503,
+        content_type="text/json",
+    )
+
+    try:
+        actual = get_request_response(test_url)
+        assert actual is None
+    except requests.exceptions.RetryError:
+        pass
+
+
+@httpretty.activate(allow_net_connect=False)
+def test_get_response_statusError_NumberOfRequests():
+    """Tests get_response number of requests code when there is an error"""
+    test_url = "http://foo-api.com/data?page=2"
+    httpretty.register_uri(
+        httpretty.GET,
+        test_url,
+        body='{}',
+        status=503,
+        content_type="text/json",
+    )
+
+    try:
+        get_request_response(test_url)
+        assert len(httpretty.latest_requests()) == 4
```

Comment:
this should be outside of the try/except. line 95 throws the exception which causes this line to be skipped.